src/Viewer.tsx

```diff
@@ -444,8 +445,19 @@ function Viewer(): ReactElement {
           datasetKey = newCollection.getDefaultDatasetKey();
         } else {
           // Try loading the collection, with the default collection as a fallback.
-          newCollection = await Collection.loadCollection(collectionUrlParam || DEFAULT_COLLECTION_PATH);
-          datasetKey = datasetParam || newCollection.getDefaultDatasetKey();
+          try {
+            newCollection = await Collection.loadCollection(collectionUrlParam || DEFAULT_COLLECTION_PATH);
+            datasetKey = datasetParam || newCollection.getDefaultDatasetKey();
+          } catch (error) {
+            console.error(error);
+            notificationApi["error"]({
+              message: "Error loading dataset:",
+              description: (error as Error).message,
+              placement: "bottomLeft",
+              duration: 4,
+            });
+            return;
+          }
```

Comment:
Better error handling when bad datasets are loaded in-- this will pop up an error notification.